packages/taquito-beacon-wallet/src/taquito-beacon-wallet.ts

```diff
@@ -72,16 +72,49 @@ export class BeaconWallet implements WalletProvider {
     return account.address;
   }
 
-  async mapTransferParamsToWalletParams(params: WalletTransferParams) {
-    return this.removeDefaultParams(params, await createTransferOperation(this.formatParameters(params)));
+  async mapTransferParamsToWalletParams(params: () => Promise<WalletTransferParams>) {
+    let walletParams: WalletTransferParams;
+    await this.client.showPrepare();
+    try {
+      walletParams = await params();
```

Comment:
Can you please point me where params() function is defined.

Reply:
Here for the origination for example: https://github.com/ecadlabs/taquito/blob/700ca12331bb15375993c2e7de74145a868a9651/packages/taquito/src/wallet/wallet.ts#L170